### PR TITLE
Bug/single frequency signals

### LINKED ISSUE #360: Audio Classes do not allow data for a single time/frequency
## General

but they should

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Closes #360 

### Changes proposed in this pull request:

- fix `Signal.__init__`, `Signal.freq`, and `Signal.freq_raw` for the case of passing single frequency data but not the number of samples
- Reduce code redundancy in `Signal.freq`, and `Signal.freq_raw` though shared functions `Signal._freq`

### CODE REVIEW

File: pyfar/classes/audio.py [hunk 1]
```diff
@@ -614,9 +614,9 @@ def __init__(
             # check and set n_samples
             if n_samples is None:
                 warnings.warn(
-                    "Number of time samples not given, assuming an even "
-                    "number of samples from the number of frequency bins.")
-                n_samples = (data.shape[-1] - 1)*2
+                    f"Number of samples not given, assuming {n_samples} "
+                    f"samples from {data.shape[-1]} frequency bins.")
+                n_samples = max(1, (data.shape[-1] - 1)*2)
```
Comment (tluebeck) [inline pyfar/classes/audio.py:619]: Does it make sense to write {n_samples} in the f-string here? n_samples is None. I suggest to print soemthing like "Number of samples not given, assuming the number of samples from {data.shape[-1]} frequency bins."

**Review by tluebeck — COMMENTED** [review]
Everything looks fine to me. Just one minor comment. 

But I need to update it in the complex_time_signal branch.

File: pyfar/classes/audio.py (hunk 1 above)
Comment (sikersten) [reply pyfar/classes/audio.py:619]: Good point, could be solved by changig the order of the warning and `n_samples=...`, right?

**Review by sikersten — APPROVED** [review]

File: pyfar/classes/audio.py (hunk 1 above)
Comment (f-brinkmann) [reply pyfar/classes/audio.py]: Thanks for noting. Changed the order as suggested to be able to output the assumed number of samples

**Review by tluebeck — APPROVED** [review]

**Pull request merged**